src/all/torrentio/src/eu/kanade/tachiyomi/animeextension/all/torrentio/Torrentio.kt

```diff
@@ -0,0 +1,891 @@
+                  objectTypes: ${"$"}objectTypes,
+                  searchQuery: ${"$"}searchQuery,
+                  packages: ${"$"}packages,
+                  genres: [],
+                  excludeGenres: [],
+                  releaseYear: ${"$"}releaseYear
+                }
+              ) {
+                edges {
+                  node {
+                    id
+                    objectType
+                    content(country: ${"$"}country, language: ${"$"}language) {
+                      fullPath
+                      title
+                      shortDescription
+                      externalIds {
+                        imdbId
+                      }
+                      posterUrl
+                      genres {
+                        translation(language: ${"$"}language)
+                      }
+                      credits {
+                        name
+                        role
+                      }
+                    }
+                  }
+                }
+                pageInfo {
+                  hasPreviousPage
+                  hasNextPage
+                }
+              }
+            }
+        """.trimIndent()
+    }
+
+    private fun parseSearchJson(jsonLine: String?): AnimesPage {
+        val jsonData = jsonLine ?: return AnimesPage(emptyList(), false)
+        val popularTitlesResponse = json.decodeFromString<GetPopularTitlesResponse>(jsonData)
+
+        val edges = popularTitlesResponse.data?.popularTitles?.edges.orEmpty()
+        val hasNextPage = popularTitlesResponse.data?.popularTitles?.pageInfo?.hasNextPage ?: false
+
+        val metaList = edges
+            .mapNotNull { edge ->
+                val node = edge.node ?: return@mapNotNull null
+                val content = node.content ?: return@mapNotNull null
+
+                SAnime.create().apply {
+                    url = "${content.externalIds?.imdbId ?: ""},${node.objectType ?: ""},${content.fullPath ?: ""}"
+                    title = content.title ?: ""
+                    thumbnail_url = "https://images.justwatch.com${content.posterUrl?.replace("{profile}", "s276")?.replace("{format}", "webp")}"
+                    description = content.shortDescription ?: ""
+                    val genresList = content.genres?.mapNotNull { it.translation }.orEmpty()
+                    genre = genresList.joinToString()
+
+                    val directors = content.credits?.filter { it.role == "DIRECTOR" }?.mapNotNull { it.name }
+                    author = directors?.joinToString()
+                    val actors = content.credits?.filter { it.role == "ACTOR" }?.take(4)?.mapNotNull { it.name }
+                    artist = actors?.joinToString()
+                    initialized = true
+                }
+            }
+
+        return AnimesPage(metaList, hasNextPage)
+    }
+
+    // ============================== Popular ===============================
+    override fun popularAnimeRequest(page: Int): Request {
```

Comment:
This is pretty much identical to `searchAnimeRequest`, use that instead and pass in the query as an empty string